### ISSUE TITLE
IMDSv2 support requires minio-go/v7 v7.0.13

### ISSUE THREAD

**Issue by chrisHMRC** [thanos-io/thanos#5101]
### Is your proposal related to a problem?

Current Thanos does not support AWS IMDSv2 

### Describe the solution you'd like

Upgrade the minio-go library to IMDSv2 supported version i.e v7.0.11+

### Describe alternatives you've considered

Only way.

### Additional context

Would be fantastic if it can be added asap.

**Comment by chrisHMRC** [comment]
Believe this is more complicated that just incrementing the minio version; since currently thanos is running a fork; from another issue.

**Comment by krishicks** [comment]
👋 I'd love to see IMDSv2 support added to Thanos. I cloned the upstream and found it merged cleanly into the [fork](https://github.com/bwplotka/minio-go/tree/optimize).

Is it possible to bump the fork as a quick way to get IMDSv2 support?

**Comment by vxe** [comment]
Hi do we have an ETA for this?

**Comment by bwplotka** [comment]
Sounds like more people are impacted with this. Help wanted! (:

**Comment by wiardvanrij** [comment]
We all want it fast, but we will hit new issues if we don't move back to upstream and provide upstream fixes when needed. 

The fork was basically a hotfix for a serious issue. This is, with all due respect, a feature request. 

The clean path is to move back to upstream if possible. We can't start to maintain a fork of minio (: As we are also an OSS project, this is maybe a nice incentive to get help on this issue.

P.s. I'm not strictly saying that this is the path we must walk, but from a projects point of view the most healthy path IMO.

**Comment by bwplotka** [comment]
Blocked on: https://github.com/thanos-io/thanos/issues/5393

**Comment by bwplotka** [comment]
I updated fork, so we can start using IMDSv2 unless there is anything to do on our side to enable it.

See: https://github.com/thanos-io/thanos/pull/5474

**Comment by markmsmith** [comment]
@bwplotka Did you figure out if this just works now, or if there's remaining work to be done in Thanos to enable it?

**Comment by oded-dd** [comment]
Hey. What is the current status?

**Comment by matej-g** [comment]
Hey @oded-dd, we're now on a [newer `minio-go`](https://github.com/thanos-io/objstore/blob/main/go.mod) version, so unless there are more extra steps needs to enable IMDSv2, this should be considered resolved.

**Comment by oded-dd** [comment]
@matej-g Thank you, we tested it and it seems to work for version 0.28

**Comment by matej-g** [comment]
Thanks for the confirmation @oded-dd, in that case I'm closing this.
